.eslintrc.js

```diff
@@ -0,0 +1,4 @@
+/** @module {CjsModule} */
+'use strict';
+require = require('esm')(module);
```

Comment:
this is confusing to me. you're reassigning the modules' `require` function?

it seems like it'd be much simpler to put the config in this `.js` file directly rather than needing the `esm` indirection.

Reply:
Yeah, this is a little tough to grasp. This is colloquially called "hijacking require" and there are other ways of doing it.

This is preferable because now we get to use ES modules for our config. 😄

There is another way we can do this w/o depending on the `esm` package. Have you seen [my StackOver answer](https://stackoverflow.com/questions/48624588/use-export-default-in-eslintrc-js-file-instead-of-module-exports/57641788#57641788)?

Reply:
This is not yet addressed. I know why this is not appreciated in this repo and I will fix it.

Reply:
Why does it matter that we’re using ESM for the config?

Reply:
> Why does it matter that we’re using ESM for the config?

It matters because ESM is a standard.

> What advantage does that offer?

The advantage is that we will be using spec-compliant module syntax.

> Static CJS and static ESM are conceptually identical.

These wouldn't be static. JSON is a static data interchange format.

Reply:
CJS is a standard too, in node, which is the only place this config will ever be used.

Both are static when you use them statically, and both are dynamic when used dynamically (both support `import()`, for example). ESM offers no "purity" advantage here, just added complexity.

Reply:
Okay, I will change it to use CommonJS. I thought this was leading to JSON configs, which loses comments.